### ISSUE TITLE
Incompatible with esbonio 0.16.0

### ISSUE THREAD

**Issue by ubmarco** [useblocks/sphinx-needs#867]
esbonio 0.16.0 got released on 4th of Feb 2023.
sphinx-needs uses `esbonio = ">=0.11.3"` which auto-installs 0.16.0.

This leads to
```
  File "/home/marco/test/docs/.venv/lib/python3.8/site-packages/sphinx_modeling/__init__.py", line 3, in <module>
    from sphinx_modeling.setup import setup  # needed by Sphinx to configure the extension  # noqa: F401
  File "/home/marco/test/docs/.venv/lib/python3.8/site-packages/sphinx_modeling/setup.py", line 11, in <module>
    from sphinx_needs.api import add_dynamic_function, add_extra_option, add_need_type
  File "/home/marco/test/docs/.venv/lib/python3.8/site-packages/sphinx_needs/__init__.py", line 1, in <module>
    from sphinx_needs.lsp.esbonio import esbonio_setup  # noqa: F401
  File "/home/marco/test/docs/.venv/lib/python3.8/site-packages/sphinx_needs/lsp/esbonio.py", line 10, in <module>
    from esbonio.lsp import LanguageFeature
  File "/home/marco/test/docs/.venv/lib/python3.8/site-packages/esbonio/lsp/__init__.py", line 14, in <module>
    from lsprotocol.types import COMPLETION_ITEM_RESOLVE
ModuleNotFoundError: No module named 'lsprotocol'
```
I guess constraining the upper version to `esbonio<0.16.0` is the right decision as it is [about to be removed](https://github.com/useblocks/sphinx-needs/issues/845#issue-1506297447) anyway.

**Comment by arwedus** [comment]
possible duplicate of #865

**Comment by danwos** [comment]
Closed because of duplicate of #865.
